NeoML/src/Dnn/Layers/CrossEntropyLossLayer.cpp

```diff
@@ -38,12 +38,11 @@ void CCrossEntropyLossLayer::BatchCalculateLossAndGradient( int batchSize, CCons
 void CCrossEntropyLossLayer::BatchCalculateLossAndGradient( int batchSize, CConstFloatHandle data, int vectorSize,
 	CConstFloatHandle label, int labelSize, CFloatHandle lossValue, CFloatHandle lossGradient, CFloatHandle labelLossGradient )
 {
-	NeoAssert( labelSize == vectorSize );
+	CheckArchitecture( labelSize == vectorSize, GetName(), "for float labels the dimensions should be equal to the first input dimensions" );
```

Comment:
What does `first` mean here? Because in NeoML order `batchSize` goes _prior_ to `vectorSize`. M.b. just use vector size?